pkg/client/googlegroup_github/mapper.go

```diff
@@ -19,15 +19,80 @@ import (
 	"context"
 	"fmt"
 	"os"
+	"slices"
 	"strings"
 
 	"google.golang.org/protobuf/encoding/prototext"
 
 	"github.com/abcxyz/pkg/logging"
 	"github.com/abcxyz/team-link/apis/v1alpha3"
+	"github.com/abcxyz/team-link/pkg/github"
 	"github.com/abcxyz/team-link/pkg/groupsync"
 )
 
+// GroupMapper implements groupsync.OneToManyGroupMapper
+// For now it was be used as Mapper between GoogleGroup
+// and GitHubTeam.
+// Both GoogleGroupMapper and GitHubMapper have similar
+// variables and functions, this way we don't need to repeatly
+// write similar function for both on them.
+type GroupMapper struct {
+	mappings map[string][]string
+}
+
+func (m *GroupMapper) AllGroupIDs(ctx context.Context) ([]string, error) {
+	res := make([]string, 0, len(m.mappings))
+	for key := range m.mappings {
+		res = append(res, key)
+	}
+	slices.Sort(res)
+	return res, nil
+}
+
+func (m *GroupMapper) ContainsGroupID(ctx context.Context, key string) (bool, error) {
+	_, ok := m.mappings[key]
+	if !ok {
+		return false, nil
+	}
+	return ok, nil
+}
+
+func (m *GroupMapper) MappedGroupIDs(ctx context.Context, key string) ([]string, error) {
+	x, ok := m.mappings[key]
+	if !ok {
+		return nil, fmt.Errorf("no mapping found for group ID: %s", key)
+	}
+	return x, nil
```

Comment:
I'd encourage us to make a deepcopy of the slice here. This is passing a slice by reference, so we're in for a bad time if the caller modifies the returned slice (because it'll end up modifying our slice too):

```go
x := m.MappedGroupIDs(ctx, "foo")
x = append(x, "bar") // equivalent to m.mappings[key] = append(m.mappings[key], "bar")
```

Reply:
Done. Thank you!